### PR TITLE
vmctl: disable progress bar for prometheus snapshot migrations

### PR DESCRIPTION
**Pull request by dmitryk-dk**
Disable progressbar for `prometheus` migration process via the flag `--prom-disable-progress-bar`.

Related issue: https://github.com/VictoriaMetrics/VictoriaMetrics/issues/6367

How to test

1. Create docker compose file like:
```
version: "3.5"
services:
  prometheus:
    container_name: prometheus
    image: prom/prometheus:v2.30.0
    volumes:
      - ./prometheus.yml:/etc/prometheus/prometheus.yml
    command:
      - "--config.file=/etc/prometheus/prometheus.yml"
      - "--web.enable-admin-api"
    ports:
      - 9090:9090
    networks:
      - vm_net
    restart: always

  avalanche:
    container_name: avalanche
    image: quay.io/prometheuscommunity/avalanche:main

    command:
      - "--metric-count=1000"
    ports:
      - 9001:9001
    networks:
      - vm_net
    restart: always
 
  victoriametrics:
    container_name: victoriametrics
    image: victoriametrics/victoria-metrics:v1.101.0
    ports:
      - 8428:8428
      - 8089:8089
      - 8089:8089/udp
      - 2003:2003
      - 2003:2003/udp
      - 4242:4242
    volumes:
      - vmdata:/storage
    command:
      - "--storageDataPath=/storage"
      - "--graphiteListenAddr=:2003"
      - "--opentsdbListenAddr=:4242"
      - "--httpListenAddr=:8428"
      - "--influxListenAddr=:8089"
      - "--vmalert.proxyURL=http://vmalert:8880"
    networks:
      - vm_net
    restart: always

volumes:
  vmagentdata: {}
  vmdata: {}
  grafanadata: {}
networks:
  vm_net:
```

prepare the `prometheus.yml` file like
```
global:
  scrape_interval: 10s

scrape_configs:
  - job_name: 'victoriametrics'
    static_configs:
      - targets: ['victoriametrics:8428']
  - job_name: 'avalanche'
    static_configs:
      - targets: ['avalanche:9001']
```

Wait sometime while prometheus scrapes avalanche

2. Create a snapshot `curl -XPOST http://127.0.0.1:9090/api/v1/admin/tsdb/snapshot`

3. Copy snapshot to the local machine 
```docker cp prometheus:/prometheus/data/snapshots path/to/folder```

Run without flag
```
./vmctl prometheus --prom-snapshot=/Users/dmitryk-dk/VictoriaMetrics/bin/snapshots/20240530T114857Z-43b3e75e21ddb558 \
  --vm-concurrency=1 \
  --vm-batch-size=200000 \
  --prom-concurrency=3 \
--vm-addr=http://localhost:8428/
Prometheus import mode
Prometheus snapshot stats:
  blocks found: 1;
  blocks skipped by time filter: 0;
  min time: 1717069660305 (2024-05-30T13:47:40+02:00);
  max time: 1717069730647 (2024-05-30T13:48:50+02:00);
  samples: 97090;
  series: 20890.
Found 1 blocks to import. Continue? [Y/n] y
2024/05/30 13:53:56 Import finished!
2024/05/30 13:53:56 VictoriaMetrics importer stats:
  idle duration: 0s;
  time spent while importing: 196.974333ms;
  total samples: 97090;
  samples/s: 492906.86;
  total bytes: 13.0 MB;
  bytes/s: 65.8 MB;
  import requests: 1;
  import requests retries: 0;
VM worker 0:↖ ? p/s                                                                                                                                                                                                 
Processing blocks: 1 / 1 [█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████] 100.00%
2024/05/30 13:53:56 Total time: 6.038979834s
```

Run with flag
```
./vmctl prometheus --prom-snapshot=/Users/dmitryk-dk/VictoriaMetrics/bin/snapshots/20240530T114857Z-43b3e75e21ddb558 \
  --vm-concurrency=1 \
  --vm-batch-size=200000 \
  --prom-concurrency=3 \
--vm-addr=http://localhost:8428/ \
> --prom-disable-progress-bar=true
Prometheus import mode
Prometheus snapshot stats:
  blocks found: 1;
  blocks skipped by time filter: 0;
  min time: 1717069660305 (2024-05-30T13:47:40+02:00);
  max time: 1717069730647 (2024-05-30T13:48:50+02:00);
  samples: 97090;
  series: 20890.
Found 1 blocks to import. Continue? [Y/n] y
2024/05/30 13:54:23 Import finished!
2024/05/30 13:54:23 VictoriaMetrics importer stats:
  idle duration: 0s;
  time spent while importing: 191.499958ms;
  total samples: 97090;
  samples/s: 506997.50;
  total bytes: 13.0 MB;
  bytes/s: 67.7 MB;
  import requests: 1;
  import requests retries: 0;
2024/05/30 13:54:23 Total time: 1.629871584s
```

### CODE REVIEW

**Comment by zekker6** [comment]
Wdyt about adding a global option to disable progress bar instead of an option just for Prometheus?
Similar to what was added for TTY detection [here](https://github.com/VictoriaMetrics/VictoriaMetrics/pull/3938/files).

It should make it easier for the end user to disable the progress by just adding a global flag rather than checking if the flag exists for the required migration mode.
Also, it would make sense to deprecate an existing flag with `--vm` in order to avoid confusion of having two flags doing the same.

File: app/vmctl/influx.go [hunk 1]
```diff
@@ -6,35 +6,100 @@ import (
...
+func WithVerbose(verbose bool) InfluxProcessorOption {
+	return func(ip *influxProcessor) {
+		ip.isVerbose = verbose
+	}
+}
+
+// WithDisableProgressBar sets disable progress bar for processor
+func WithDisableProgressBar(disableProgressBar bool) InfluxProcessorOption {
+	return func(ip *influxProcessor) {
+		ip.disableProgressBar = disableProgressBar
```
Comment (zekker6) [inline app/vmctl/influx.go:102]: Could you move these changes to a separate PR?
It would be better to keep a PR scope related to 1 issue.

File: docs/CHANGELOG.md [hunk 2]
```diff
@@ -71,6 +71,7 @@ See also [LTS releases](https://docs.victoriametrics.com/lts-releases/).
...
 * BUGFIX: [dashboards](https://grafana.com/orgs/victoriametrics): fix `AnnotationQueryRunner` error in Grafana when executing annotations query against Prometheus backend. See [this issue](https://github.com/VictoriaMetrics/VictoriaMetrics/issues/6309) for details.
 * BUGFIX: [Single-node VictoriaMetrics](https://docs.victoriametrics.com/) and `vmselect` in [VictoriaMetrics cluster](https://docs.victoriametrics.com/cluster-victoriametrics/): filter deleted label names and values from [`/api/v1/labels`](https://docs.victoriametrics.com/url-examples/#apiv1labels) and [`/api/v1/label/.../values`](https://docs.victoriametrics.com/url-examples/#apiv1labelvalues) responses when `match[]` filter matches small number of time series. The issue was introduced [v1.81.0](https://docs.victoriametrics.com/changelog_2022/#v1810).
+* BUGFIX: [vmctl](https://docs.victoriametrics.com/vmctl/): add `--disable-progress-bar` global command-line flags for all protocols. It can be used for disabling progress bar per each worker during the import. flag `--vm-disable-progress-bar` will be DEPRECATED in the future. See [this issue](https://github.com/VictoriaMetrics/VictoriaMetrics/issues/6367).
```
Comment (zekker6) [inline docs/CHANGELOG.md:74]: Could you also add deprecation notice to the top of `tip` section into as an "update note"?

```suggestion
* BUGFIX: [vmctl](https://docs.victoriametrics.com/vmctl/): add `--disable-progress-bar` global command-line flag. It can be used for disabling dynamic progress bar for any migration modes.  `--vm-disable-progress-bar`  command-line flag is deprecated and will be removed in the future releases.  See [this issue](https://github.com/VictoriaMetrics/VictoriaMetrics/issues/6367).
```

File: app/vmctl/influx.go (hunk 1 above)
Comment (dmitryk-dk) [reply app/vmctl/influx.go]: rollbacked, added just needed param

File: docs/CHANGELOG.md (hunk 2 above)
Comment (dmitryk-dk) [reply docs/CHANGELOG.md]: updated

File: app/vmctl/flags.go [hunk 3]
```diff
@@ -10,8 +10,9 @@ import (
 )
 
 const (
-	globalSilent  = "s"
-	globalVerbose = "verbose"
+	globalSilent             = "s"
+	globalVerbose            = "verbose"
+	globalDisableProgressBar = "disable-progress-bar"
```
Comment (hagen1778) [inline app/vmctl/flags.go:15]: I think this flag needs to be passed to `barpool` subpackage. When set, all operations with bar needs to be no-op. This would simplify bar usage for all other packages.

File: app/vmctl/flags.go [hunk 4]
```diff
@@ -10,8 +10,9 @@ import (
 )
 
 const (
-	globalSilent  = "s"
-	globalVerbose = "verbose"
+	globalSilent             = "s"
```
Comment (hagen1778) [inline app/vmctl/flags.go:13]: I noticed that some packages suppress progress bar if `s` is set. This looks incorrect. Let's put an explicit description for these flags:
* s - suppresses all prompts
* verbose - enables additional log messages
* disable-progress-bar - disables the progress bar during migration

This logic defined above need to be true for all migration methods.

**Review by hagen1778 — CHANGES_REQUESTED** [review]

File: app/vmctl/flags.go (hunk 3 above)
Comment (dmitryk-dk) [reply app/vmctl/flags.go:15]: Use this flag as a param when initializing progress bar

File: app/vmctl/flags.go (hunk 4 above)
Comment (dmitryk-dk) [reply app/vmctl/flags.go:13]: Disable progress bar only if `disable-progress-bar` flag is used

**Comment by hagen1778** [comment]
@dmitryk-dk would you mind reviewing my last commits? My intention was to simplify usage of prompts and progress bar.

**Comment by dmitryk-dk** [comment]
> @dmitryk-dk would you mind reviewing my last commits? My intention was to simplify usage of prompts and progress bar.

Hi @hagen1778 ! Checked on my local and it works as expected

**Review by hagen1778 — APPROVED** [review]
